tests/yast2_gui/yast2_firewall_libyui.pm

```diff
@@ -0,0 +1,207 @@
+# SUSE's openQA tests
+#
+# Copyright 2009-2013 Bernhard M. Wiedemann
+# Copyright 2012-2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: yast2-firewall yast2-http-server apache2 apache2-prefork firewalld
+# Summary: YaST2 Firewall UI test checks verious configurations and settings of firewall
+# Make sure yast2 firewall can opened properly. Configurations can be changed and written correctly.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils qw(is_sle is_leap is_tumbleweed);
+use yast2_shortcuts '%fw';
+use network_utils 'iface';
+use yast2_widget_utils 'change_service_configuration';
+use YaST::Module;
+
+sub susefirewall2 {
+    # 	enter page interfaces and change zone for network interface
+    assert_and_click("yast2_firewall_config_list");
+    assert_screen "yast2_firewall_interfaces";
+    assert_and_click("yast2_firewall_interface_zone_change");
+    wait_still_screen(2);
+    assert_and_click("yast2_firewall_interface_no-zone_assigned");
+    wait_still_screen 1;
+    wait_screen_change {
+        send_key "down";
+        send_key "ret";
+    };
+    wait_still_screen 1;
+    send_key "alt-o";
+    assert_screen "yast2_firewall_interfaces";
+
+    # 	enter page Allowed Services and make  some changes
+    assert_and_click("yast2_firewall_allowed-services");
+    assert_and_click("yast2_firewall_service-to-allow");
+    assert_and_click("yast2_firewall_service_http");
+    send_key "alt-a";
+    assert_screen "yast2_firewall_service_http_addded";
+
+    #	enter page Broadcast and disable logging broadcast packets
+    assert_and_click("yast2_firewall_broadcast");
+    wait_still_screen 1;
+    wait_screen_change { send_key "alt-l"; };
+    send_key "alt-o";
+    assert_screen "yast2_firewall_broadcast_no-logging";
+
+    # 	enter page Logging Level and disable logging
+    assert_and_click("yast2_firewall_logging-level");
+    assert_and_click("yast2_firewall_do-not-log-any_accepted");
+    assert_and_click("yast2_firewall_do-not-log-any_not-accepted");
+
+    #	enter page Custom Rules and check ui
+    assert_and_click("yast2_firewall_custom-rules");
+    # verify Custom Rules page is displayed
+    assert_screen("yast2_firewall_custom-rules-loaded");
+    send_key "alt-a";
+    assert_screen "yast2_firewall_add-new-allowing-rules";
+    send_key "alt-c";
+    wait_still_screen(2);
+
+    #	Next to finish and exit
+    send_key "alt-n";
+    assert_screen "yast2_firewall_summary", 30;
+    send_key "alt-f";
+}
+
+sub verify_service_stopped {
+
+    record_info('Start-Up', "Managing the firewalld service: Stop");
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    assert_screen 'yast2_firewall_start-up';
```

Comment:
we should not need this needle at all, just check that an element of the page where we land exists.